src/typing_extensions.py

```diff
@@ -877,6 +878,59 @@ def inner(func):
     return inner
 
 
+if not hasattr(typing, "NoDefault") or not hasattr(typing, "NoExtraItems"):
+    class SingletonMeta(type):
```

Comment:
kind of surprised we bothered. tbh if someone wants to do type(NoDefault).asdf = 123 i could live with that

Reply:
I guess I did it initially for parity with the C version in CPython for NoDefault, where it's easier to not accept extra attributes, and now that I need another one it makes sense to follow the pattern.